features/fields-fout-cases.feature

```diff
@@ -0,0 +1,140 @@
+    | burgerservicenummer | 999994086                       |
+    | fields              |                                 |
+    Dan heeft de response een object met de volgende gegevens
+    | naam     | waarde                                                                                                      |
+    | type     | https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?#System_Net_HttpStatusCode_BadRequest |
+    | title    | Een of meerdere parameters zijn niet correct.                                                               |
+    | status   | 400                                                                                                         |
+    | detail   | De foutieve parameter(s) zijn: fields.                                                                      |
+    | code     | paramsValidation                                                                                            |
+    | instance | /haalcentraal/api/brp/personen                                                                              |
+    En heeft het object de volgende 'invalidParams' gegevens
+    | code    | name   | reason                                              |
+    | pattern | fields | Waarde voldoet niet aan patroon ^[a-zA-Z0-9\.,_]+$. |
+
+Rule: De Fields parameter bevat het volledige pad naar de op te vragen velden gescheiden door een komma
+
+  @fout-case
+  Scenario: Gevraagde veld bestaat niet
+    Als personen wordt gezocht met de volgende parameters
+    | naam                | waarde                          |
+    | type                | RaadpleegMetBurgerservicenummer |
+    | burgerservicenummer | 999994086                       |
+    | fields              | burgerservicenummer,bestaatniet |
+    Dan heeft de response een object met de volgende gegevens
+    | naam     | waarde                                                                                                      |
+    | type     | https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?#System_Net_HttpStatusCode_BadRequest |
+    | title    | Een of meerdere parameters zijn niet correct.                                                               |
+    | status   | 400                                                                                                         |
+    | detail   | De foutieve parameter(s) zijn: fields.                                                                      |
+    | code     | paramsValidation                                                                                            |
+    | instance | /haalcentraal/api/brp/personen                                                                              |
+    En heeft het object de volgende 'invalidParams' gegevens
+    | code   | name   | reason                                       |
+    | fields | fields | Parameter bevat een niet bestaande veldnaam. |

```

Comment:
@MelvLee De implementatie levert nu: "Deel van de parameterwaarde niet correct: bestaatniet."

Wil je dat gewijzigd zien?

Reply:
Deze wijziging is gedaan tbv consistentie. In andere foutmeldingen wordt niet gecommuniceerd wat binnen de parameter waarde niet voldoet

Reply:
ik ben alleen bang dat bij veel velden in fields het erg moeilijk is voor een gebruiker te achterhalen welk veld het probleem geeft

Reply:
Bij veel velden gaat het waarschijnlijk ook niet passen in een reason van max. 80 karakters

Reply:
klopt. De langste zou zijn "Deel van de parameterwaarde niet correct:  kinderen.geboorte.inOnderzoek.datumIngangOnderzoek.onbekend", van 102 tekens in totaal (incl. de foutmelding tekst).

Is er een reden om reason dan niet uit te breiden om een langere tekst toe te laten?

In theorie kan iemand de maximale lengte van de fieldsparameter, 924 tekens (waarom dat aantal?), fout typen. Maar ik denk dat dat technisch afgevangen kan worden, bijvoorbeeld door op een gegeven moment te eindigen met "...".

@MelvLee ben je het niet eens met mijn gedachte dat het voor een gebruiker (developer) handig is om terug te krijgen wat fout is ingevoerd?

N.B. wanneer de invoer in fields niet voldoet aan pattern, moet uiteraard de pattern fout en reason worden teruggegeven en niet de foute invoer.

Reply:
Ik twijfel omdat ik mij afvraag wat de toegevoegde waarde is. In principe gaat een developer de fields niet zelf typen omdat de fields uit de fields-mapping-*.csv bestanden kan worden gecopieerd. Hoe vaak gaat het dan voorkomen dat je het fout copy/paste?

Ook moet er logica worden toegevoegd voor de opmaak van de reason string. Wat ik nu kan bedenken:
- te lange reason afkappen met ...
- field part voldoet niet aan pattern
- field part voldoet niet aan casing
- foute field part waarde wordt meerdere keren opgegeven

Deze requirement vergroot verder de kans op script injection. De input wordt namelijk gecopieerd naar de output. Dat betekent dat er moet worden gemitigeerd. Niet alleen bij de proxy, maar ook bij alle andere intermediate endpoints. En het klopt dat we hiervoor een pattern hebben bedacht, maar het moet dan wel getest worden dat het ook wordt gebruikt in de code

En mogelijk wordt fields een array ipv een komma seperated string. Dan veranderen waarschijnlijk ook de requirements voor fields

Reply:
oke, ik heb het voor nu aangepast naar jouw voorstel: "Parameter bevat een niet bestaande veldnaam."